### ISSUE TITLE
Dynamips error when running command 'frsw create_vc "FR1"

### ISSUE THREAD

**Issue by Khriton** [GNS3/gns3-gui#2051]
I don't see this issue listed anywhere else, so here goes.

Windows 10
GNS3 2.0.0 (release version from https://www.gns3.com/software/download)
GNS VM 0.10.14 (https://www.gns3.com/software/download-vm)

I am able to configure the frame relay switch with 4 ports.  I am able to connect to any two (rarely 3) ports of my choice, then I get the following errors:

Error while creating link: Dynamips error when running command 'frsw create_vc "FR1" udp-c3d12045-8375-43a5-b9ad-0a62870c2095 102 udp-82d54176-2100-448c-af36-526a12966e80 201
': unable to create VC
Error while creating link: Dynamips error when running command 'frsw create_vc "FR1" udp-c3d12045-8375-43a5-b9ad-0a62870c2095 102 udp-82d54176-2100-448c-af36-526a12966e80 201
': unable to create VC

Further attempts to connect result in:
Error while creating link: Port 3 isn't free
Error while creating link: Port 4 isn't free

I tried:
Closed out and rebooted GNS3
Uninstalled and reinstalled GNS3 and the various components
New projects
Using my machine as the server (gives same errors)
GNS3 Doctor says everything is OK
Layout is a frame relay switch with four c3600 routers (c3660-jk9o3s-mz.124-25d.image)

I am not sure how popular frame relay is, but I only found one search result on Reddit for the same issue.  Please let me know if I can provide any additional information.

**Comment by noplay** [comment]
I suspect a duplicate of: https://github.com/GNS3/gns3-server/issues/1005

Thanks for the additionnal informations

**Comment by grossmj** [comment]
I've tested Frame Relay in 2.0 and found not issues.

For reference:

![frame_relay](https://cloud.githubusercontent.com/assets/4752614/25935400/9a2b1308-3655-11e7-8729-c8c2d03d3c16.png)

![frame_relay_2](https://cloud.githubusercontent.com/assets/4752614/25935404/a10b5bce-3655-11e7-951f-e8e5da827171.png)

R1 config:

```
interface Serial1/0
 no ip address
 encapsulation frame-relay
 serial restart-delay 0
!
interface Serial1/0.1 point-to-point
 ip address 10.0.0.1 255.0.0.0
 snmp trap link-status
 frame-relay interface-dlci 101  
```

R2 config:

```
interface Serial1/0
 no ip address
 encapsulation frame-relay
 serial restart-delay 0
!
interface Serial1/0.1 point-to-point
 ip address 10.0.0.2 255.0.0.0
 snmp trap link-status
 frame-relay interface-dlci 202  
```

**Comment by grossmj** [comment]
Trying to connect more than 2 devices to a Frame Relay switch leads to errors:

```
Error while creating link: Dynamips error when running command 'frsw create_vc "FR1" udp-4328df48-0536-443b-a685-1b04bcb4aff2 101 udp-492d9447-dd00-4c07-ad9c-ffb5f0128b67 202
': unable to create VC
Error while creating link: Dynamips error when running command 'frsw create_vc "FR1" udp-f2054ced-c520-46e0-b061-49d323aba88e 101 udp-77fbf023-1001-40bf-837b-ec0b06c3de86 202
': unable to create VC
Error while creating link: Port 2 isn't free
```

**Comment by grossmj** [comment]
This is a regression. It works in 1.5.x.

**Comment by grossmj** [comment]
Frame-relay VC mappings were not recorded properly. Fixed in https://github.com/GNS3/gns3-server/commit/87d42f4eb9fa8770071b5eae5abc97a1dbc38bee

**Comment by CablingOptional** [comment]
I'm encountering the same error in GNS 3 2.0.0 on Windows 10 64-bit.
In my case, the routers and FR1 switch are all running in a VM.

I see the comment that this has been fixed, but when I run 'Check for Update' from the app, it says I'm running the latest build.

Tips on getting the fixed version?

Thanks

**Comment by grossmj** [comment]
This issue has been fixed in version 2.0.1

**Comment by CablingOptional** [comment]
Oh geez - I see that now.  Thanks so much for all you and your team's hard
work on this excellent project.
